### ISSUE TITLE
[Feature] Visualize Type, MethodInfo

### ISSUE THREAD

**Issue by sdanyliv** [agileobjects/ReadableExpressions#42]
Hi, 
Thanks for great extension. It really helps me in modifying and analyzing expression trees.

But one thing i'm missing. Visual Studio 2019 and previous versions shows in hint very uninformative information for types, especially for complex generic types.

It would be better to see instead of
```
System.Tuple`2[[Tests.Playground.EagerLoadingTests+MasterClass, linq2db.Tests.Playground, Version=2.6.0.0, Culture=neutral, PublicKeyToken=e41013125f9e410a],[Tests.Playground.EagerLoadingTests+DetailClass, linq2db.Tests.Playground, Version=2.6.0.0, Culture=neutral, PublicKeyToken=e41013125f9e410a]], mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
```
Shorter version
```
Tuple<MasterClass, DetailClass>
```

Also during modification Expression Tree, there are many `MakeGenericMethod` calls, so it is also better to show shorter hint for `MethodInfo`
Instead of 
```
{System.Linq.IQueryable`1[System.Tuple`2[Tests.Playground.EagerLoadingTests+MasterClass,Tests.Playground.EagerLoadingTests+DetailClass]] SelectMany[MasterClass,DetailClass,Tuple`2](System.Linq.IQueryable`1[Tests.Playground.EagerLoadingTests+MasterClass], System.Linq.Expressions.Expression`1[System.Func`2[Tests.Playground.EagerLoadingTests+MasterClass,System.Collections.Generic.IEnumerable`1[Tests.Playground.EagerLoadingTests+DetailClass]]], System.Linq.Expressions.Expression`1[System.Func`3[Tests.Playground.EagerLoadingTests+MasterClass,Tests.Playground.EagerLoadingTests+DetailClass,System.Tuple`2[Tests.Playground.EagerLoadingTests+MasterClass,Tests.Playground.EagerLoadingTests+DetailClass]]])}
```
More informative version
```
IQueryable<Tuple<MasterClass, DetailClass>> SelectMany<MasterClass, DetailClass, Tuple<MasterClass, DetailClass>>
(
	IQueryable<MasterClass> source,
	Expression<Func<MasterClass, IEnumerable<DetailClass>>> collectionSelector,
	Expression<Func<MasterClass, DetailClass, Tuple<MasterClass, DetailClass>>> resultSelector
)
```
I have did `MethodInfo` visualization myself using assembly level `DebuggerDisplay`
```cs
[assembly: DebuggerDisplay("{LinqToDB.Common.Internal.MethodInfoExtensions.ShortDisplayName(this)}", Target = typeof(MethodInfo))]
```
But with no chance to visualize `Type`, the following attribute won't work
```cs
[assembly: DebuggerDisplay("{LinqToDB.Common.Internal.TypeExtensions.ShortDisplayName(this)}", Target = typeof(Type))]
```

It is possible to visualize `Type` and `MethodInfo` using your extension?
Thanks.

**Comment by SteveWilkes** [comment]
Hi!

Thanks for the feedback, and I'm really glad you've found the extension helpful :)

I could definitely setup a Type visualizer - ReadableExpressions has a public extension method on Type called [GetFriendlyName](https://github.com/agileobjects/ReadableExpressions/blob/master/ReadableExpressions/Extensions/PublicTypeExtensions.cs) - and I don't *think* it'd be especially difficult to use that to visualize a MethodInfo.

I'll check it out soon and get back to you. Thanks for the suggestion!

Cheers,

Steve

**Comment by sdanyliv** [comment]
Thanks!
Really missing this features. I'm spending several weeks in modifying very complicated linq expressions and it is annoying to get `Can not convert` exception and then analyse without visualizers what i have missed.

And again, thanks for great extension.

**Comment by SteveWilkes** [comment]
Ok, the latest code providers visualization for Type, MethodInfo, ConstructorInfo, Func and Action - release to follow.

I might split it out into a separate project (ReadableReflection?) in a subsequent release. Thanks for the suggestion! :)

**Comment by SteveWilkes** [comment]
This is now available in [v2.4 of the Visualizers](https://marketplace.visualstudio.com/items?itemName=vs-publisher-1232914.ReadableExpressionsVisualizers). Hope it's helpful, and thanks again for the suggestion!
